src/include/migraphx/type_name.hpp

```diff
@@ -35,8 +35,24 @@ std::string compute_type_name()
 {
     std::string name;
 #if defined(_MSC_VER) && !defined(__clang__)
-    name = typeid(PrivateMigraphTypeNameProbe).name();
-    name = name.substr(7);
+    const char struct_name[]    = "struct ";
+    const char class_name[]     = "class ";
+    const char function_name[]  = "compute_type_name<";
+    const char parameter_name[] = ">(void)";
+    const char cdecl_name[]     = "__cdecl";
+
+    name = __FUNCSIG__;
+
+    auto begin  = name.find(function_name) + sizeof(function_name) - 1;
+    auto length = name.find(parameter_name) - begin;
+    name        = name.substr(begin, length);
+    if(starts_with(class_name))
```

Comment:
Shouldn't this be `starts_with(name, class_name)`? I dont really understand how this even compiles.

Reply:
Yes, thanks for noticing. Why this has not generated a compile error?

Reply:
Oh, we do not use MSVC to compile MIGraphX. I will stip out this whole section.

Reply:
Yes, thanks for noticing. Why this has not generated a compile error?